packages/block/src/header.ts

```diff
@@ -525,35 +501,29 @@ export class BlockHeader {
     const maxGasLimit = parentGasLimit + a
     const minGasLimit = parentGasLimit - a
 
-    const result =
-      gasLimit < maxGasLimit &&
-      gasLimit > minGasLimit &&
-      gasLimit >= this._common.paramByHardfork('gasConfig', 'minGasLimit', hardfork)
+    if (gasLimit >= maxGasLimit) {
+      const msg = this._errorMsg('gas limit increased too much')
+      throw new Error(msg)
+    }
 
-    return result
+    if (gasLimit <= minGasLimit) {
+      const msg = this._errorMsg('gas limit decreased too much')
+      throw new Error(msg)
+    }
+
+    if (gasLimit < this._common.paramByHardfork('gasConfig', 'minGasLimit', hardfork)) {
+      const msg = this._errorMsg(
+        `gas limit decreased below minimum gas limit for hardfork=${hardfork}`
+      )
+      throw new Error(msg)
+    }
   }
 
   /**
-   * Validates the block header, throwing if invalid. It is being validated against the reported `parentHash`.
-   * It verifies the current block against the `parentHash`:
-   * - The `parentHash` is part of the blockchain (it is a valid header)
-   * - Current block number is parent block number + 1
-   * - Current block has a strictly higher timestamp
-   * - Additional PoW checks ->
-   *   - Current block has valid difficulty and gas limit
-   *   - In case that the header is an uncle header, it should not be too old or young in the chain.
-   * - Additional PoA clique checks ->
-   *   - Various extraData checks
-   *   - Checks on coinbase and mixHash
-   *   - Current block has a timestamp diff greater or equal to PERIOD
-   *   - Current block has difficulty correctly marked as INTURN or NOTURN
-   * @param blockchain - validate against an @ethereumjs/blockchain
-   * @param height - If this is an uncle header, this is the height of the block that is including it
+   * Checks static parameters related to consensus algorithm
+   * @throws if any check fails
    */
-  async validate(blockchain: Blockchain, height?: bigint): Promise<void> {
-    if (this.isGenesis()) {
-      return
-    }
+  _consensusFormatValidation() {
```

Comment:
Can you put this method upper in the method order and place it directly under `_genericFormatValidation()`? Looks a bit misplaced here.